src/vm/threads.cpp

```diff
@@ -7317,10 +7322,13 @@ BOOL Thread::SetStackLimits(SetStackLimitScope scope)
         }
 
         // Compute the limit used by EnsureSufficientExecutionStack and cache it on the thread. The limit
-        // is currently set at 50% of the stack, which should be sufficient to allow the average Framework
-        // function to run, and to allow us to throw and dispatch an exception up a reasonable call chain.
-        m_CacheStackSufficientExecutionLimit = reinterpret_cast<UINT_PTR>(m_CacheStackBase) - 
-            (reinterpret_cast<UINT_PTR>(m_CacheStackBase) - reinterpret_cast<UINT_PTR>(m_CacheStackLimit)) / 2;
+        // is should be sufficient to allow the average Framework function to run, and to allow us to throw
```

Comment:
I know that this is the description for EnsureSufficientExecutionStack from MSDN, but I think it is confusing and it does not match what EnsureSufficientExecutionStack is used for in practice. What it should actually do is to ensure that there is enough stack for typical non-recursive call chain to execute, including potential exception handling or garbage collection.